com.googlecode.efactory/src/com/googlecode/efactory/values/EfactoryValueConverterService.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2013 Michael Vorburger.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ *     Umesh - initial API and implementation
+ ******************************************************************************/
+package com.googlecode.efactory.values;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.xtext.conversion.IValueConverter;
+import org.eclipse.xtext.conversion.ValueConverter;
+import org.eclipse.xtext.conversion.ValueConverterException;
+import org.eclipse.xtext.nodemodel.INode;
+
+import com.googlecode.efactory.conversion.TerminalConverters;
+
+
+public class EfactoryValueConverterService

```

Comment:
Why are we creating a new class here, which extends TerminalConverters, which isn't used anywhere else, instead of just adding the new StringId value converter to the existing TerminalConverters. That would be clearer - I'll change that.